### ISSUE TITLE
Make redoc RTL support

### ISSUE THREAD

**Issue by raha1923** [Redocly/redoc#1002]
Issue type: feature

Redoc is one of the best opensource api-docs that i have found. We(baninab.co) need an api-doc in one of our projects. What we need is an api-doc that support RTL and has console to test api's, so i decided to talk to @RomanHotsiy and find know how do you think about make this tool RTL support?
We are ready to start develop redoc to make it RTL support and publish it, but i wanted to ask you to know do you have any plan to RTL support of redoc?

**Comment by raha1923** [comment]
There are a lot of projects that need to a good api-doc that can support RTL in my country and even in Asia. So i think it's a good idea to add RTL support to this tool to make it more useful in some countries that don't have LTR languages.

Could this issue marked as `feature` or even add a milestone to this project and assign this issue to that milestone?

**Comment by RomanHotsiy** [comment]
It may be a good addition.

**Comment by nanov** [comment]
This is actually a great idea, and not awfully hard to implement.

I may take a shot when I have a day to spend on it.

**Comment by raha1923** [comment]
I'm so glad because of that you have accept it as a feature @RomanHotsiy, thank you.

RTL supporting is not mirroring the content, it needs more actions and implementations than just mirroring the content, because of typography attributes and html structure and lots of this type of issues ([GOOD REFERENCE](https://medium.muz.li/web-design-for-right-to-left-languages-the-basics-287329d508cf)).

The mentioned reference i think is really necessary to read for anyone who like and want to help.

**Comment by nanov** [comment]
Would you like to help? ;)

As a person who grew up ( 15 years ) in a RTL country I do know the complexity of the task, any help would be appreciated.

**Comment by raha1923** [comment]
yes i do, I will help you to improve this product and add RTL support
feature.

**Comment by raha1923** [comment]
I'm going to work on rtl version of redoc and just because I'm redoc contributor so i have to work on it on my forked version of redoc, you can find it in [https://github.com/raha1923/redoc](here).

I'll be happy if you help me :)

**Comment by raha1923** [comment]
I think it's a good idea to set direction in theme (default value: LTR) and let users to change the theme direction using the theme configuration. In this case we can easily detect what type of languages user are using on his page (as main language) and we can set the proper custom style rules.

**Comment by raha1923** [comment]
hi there.
I need to add a CSS file to the project to write general RTL rules there and import the CSS file into the project. what is the best way to do this?
@RomanHotsiy @nanov

**Comment by raha1923** [comment]
@RomanHotsiy 
hi roman, I thought you have forgotten this issue and the related pull-request so I just decided to mention you and ask about the progress on this issue and the pull request. please let me know if there is a problem.

**Comment by RomanHotsiy** [comment]
Hey @raha1923. I do remember about this but it's big PR which requires careful review. I am a bit busy last months so sorry for the delay.

I will try to get back to it mid Feb.

**Comment by raha1923** [comment]
@RomanHotsiy good to hear from you, thank you for your attention.

Another thing roman, our company needs an API-console on Redoc, so we have to keep developing on redoc and try to implement a proper API-console on redoc. It would be so great if you help me to find the proper way to implement it on redoc. I'll add an issue about implementing API-console in the Redoc project.

**Comment by raha1923** [comment]
Well I've checked #53 and found out there is no need to add a new issue, I can use it and put additional info on it.

**Comment by saeidmh83** [comment]
@RomanHotsiy 
hi roman, please review @raha1923 Pull-request and add this feature as soon as possible.
thanks a lot
